### PR TITLE
[BEAM-2741] update JavaDoc for BeamRecord

### PR DESCRIPTION
**Pull request by XuMingmin**
R: @takidau @xumingming 

Ps, the change of `BeamSqlUdfExpression` is to address the question in #3707

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java [hunk 1]
```diff
@@ -72,110 +82,213 @@ private void addField(int index, Object fieldValue) {
...
+   * Get a {@link Boolean} value by field index, {@link ClassCastException} is thrown
+   * if type doesn't match.
+   */
   public Boolean getBoolean(int idx) {
     return (Boolean) getFieldValue(idx);
   }
 
+  /**
+   * Return the size of data fields.
+   */
```
Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java]: We were going to rename size() to getFieldCount(), right?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java [hunk 2]
```diff
@@ -47,24 +55,36 @@ public void validateValueType(int index, Object fieldValue)
...
+   /**
+    * Find the index of a given field.
+    */
    public int findIndexOfField(String fieldName){
      return fieldNames.indexOf(fieldName);
    }
 
+   /**
+    * Return the count of fields.
+    */
```
Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecordType.java]: Rename to getFieldCount()?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java (hunk 1 above)
Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java]: yes, a rebase would update it if #3702 is merged first.

**Comment by takidau** [comment]
Merged, feel free to close. Thank you!

**Comment by XuMingmin** [comment]
close #3710 thank you @takidau !

**Pull request closed without merging**